### PR TITLE
[API-56] Refactor docker related files

### PR DESCRIPTION
**Pull request by josepostiga**
## Description
This pull request addresses issue [API-56](https://github.com/vostpt/api/issues/56).

## Task items:
- [x] Fixed issue 56;
- [x] Aggregated all services related configuration inside a single, explicit, folder (.docker);
- [x] Added xDebug configuration;
- [x] Optimized PHP-FPM config;
- [x] updated docker-compose file with a better structure.

## Requirements (optional)
You should remove the previous compiled images so that the new Dockerfile gets built.

### CODE REVIEW

File: .docker/nginx/conf.d/api.conf [hunk 1]
```diff
@@ -6,7 +6,7 @@ server {
     error_log  /var/log/nginx/error.log;
     access_log /var/log/nginx/access.log;
 
-    root /var/www/api.vost.test/public;
+    root /var/www/html/api.vost.test/public;
```
Comment (quetzyg) [inline .docker/nginx/conf.d/api.conf]: Leave the path as it was. There probably isn't HTML in this project anyway.

File: .docker/php/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+FROM php:7.2.21-fpm
```
Comment (quetzyg) [inline .docker/php/Dockerfile]: Keep the PHP version as it was. That's what we currently have on the server.

File: .docker/php/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    && printf "\n" | pecl install mcrypt-1.0.2 \
+    && printf "\n" | pecl install xdebug-2.7.2 \
+    && docker-php-ext-enable mcrypt xdebug
+
+COPY conf/xdebug.ini /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
+COPY conf/zz-docker.conf /usr/local/etc/php-fpm.d/zz-docker.conf
+
+COPY --from=composer /usr/bin/composer /usr/bin/composer
+
+WORKDIR /var/www/html/api.vost.test
```
Comment (quetzyg) [inline .docker/php/Dockerfile]: Get rid of the `html` subdir from the `WORKDIR`.

File: storage/app/public/.gitignore [hunk 4]
```diff
@@ -1,2 +0,0 @@
-*
-!.gitignore
```
Comment (quetzyg) [inline storage/app/public/.gitignore]: Why the removal?

Comment (josepostiga) [reply storage/app/public/.gitignore]: It was an honest mistake.

**Comment by josepostiga** [comment]
@quetzyg I've addressed your changes requests. Care to make a final review/merge?

File: storage/app/public/.gitignore (hunk 4 above)
Comment (quetzyg) [reply storage/app/public/.gitignore]: No sweat, mate.

**Pull request merged**